spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreReactiveRepository.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import org.springframework.data.repository.reactive.ReactiveCrudRepository;
+
+/**
+ * @author Dmitry Solomakha
+ *
+ * @since 1.2
+ */
+public class FirestoreReactiveRepository implements ReactiveCrudRepository {
+
+	private FirestoreTemplate firestoreTemplate;
+
+	private Class type;
+
+	public FirestoreReactiveRepository(FirestoreTemplate firestoreTemplate, Class type) {
+		this.firestoreTemplate = firestoreTemplate;
+		this.type = type;
+	}
+
+	@Override
+	public Mono save(Object entity) {
```

Comment:
All `Mono`/`Flux` should be parameterized with real types they are wrapping. Otherwise we'll be causing warnings in downstream projects.